src/rules/relative-font-units.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * @fileoverview Enforce the use of relative units for font size.
+ * @author Tanuj Kanti
+ */
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @import { CSSRuleDefinition } from "../types.js"
+ * @typedef {"allowedFontUnits"} RelativeFontUnitsMessageIds
+ * @typedef {[{allow?: string[]}]} RelativeFontUnitsOptions
+ * @typedef {CSSRuleDefinition<{ RuleOptions: RelativeFontUnitsOptions, MessageIds: RelativeFontUnitsMessageIds}>} RelativeFontUnitsRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+const relativeFontUnits = [
+	"%",
+	"cap",
+	"ch",
+	"em",
+	"ex",
+	"ic",
+	"lh",
+	"rcap",
+	"rch",
+	"rem",
+	"rex",
+	"ric",
+	"rlh",
+];
+
+const fontSizeIdentifiers = new Set([
+	"xx-small",
+	"x-small",
+	"small",
+	"medium",
+	"large",
+	"x-large",
+	"xx-large",
+	"smaller",
+	"larger",
```

Comment:
There are more valid identifiers as per https://developer.mozilla.org/en-US/docs/Web/CSS/font-size#syntax
```suggestion
	"large",
	"x-large",
	"xx-large",
	"xxx-large",
	"smaller",
	"larger",
	"math",
	"inherit",
	"initial",
	"revert",
	"revert-layer",
	"unset"
```